### DISCUSSION TITLE
Tell the mobile app to ignore past photos

### DISCUSSION THREAD

**Discussion by vitobotta** [immich-app/immich#5582] [Q&A]
Hi! I did a first bulk import of my iCloud library to Immich using the CLI, then I installed the app on my iPhone and it's now downloading files from iCloud (that it doesn't have locally) and uploading to Immich, which is mostly a no op because I imported everything already. How can I speed this up? It seems it's gonna take ages. Is it possible to somehow tell the app to "assume" that all photos up to now have already been imported? Thanks!

**Comment by shenlong-tanwen** [comment]
The recent version of the app [(1.90)](https://github.com/immich-app/immich/releases/tag/v1.90.0) added a toggle to disable photos from iCloud from getting backed up. 

Are the photos actually getting duplicated in the timeline? The way backup is implemented in the mobile app is to actually upload the assets to the server to find the duplicates. This results in the assets getting uploaded to the server to check if they already exist even if they are already imported through some other way. We are planning to refactor this to use the hash of assets instead.

**Reply by vitobotta** [reply]
The hash method would be awesome, sounds like what the CLI does? If I disable the backup from iCloud will it still back up the photos in the camera roll?

**Reply by vitobotta** [reply]
I toggled the ignore iCloud thing but it still says 48K+ remaining like before. What am I missing?

**Reply by shenlong-tanwen** [reply]
> I toggled the ignore iCloud thing but it still says 48K+ remaining like before. What am I missing?

If the files are also available in the Camera roll, i.e, in the local device, they'd be tried irrespective of them being in iCloud or not. The toggle is for assets that are only available in iCloud and not on the device's local storage. Sadly, the best thing to do for you now is to wait till all the assets are tried and the state is updated in the mobile app

**Reply by vitobotta** [reply]
Understood, thanks. I was hoping I could skip this as it's gonna take a long time. If I knew that the mobile app has to upload the media anyway, I wouldn't have done the bulk import with the CLI in first place. 2.5 days wasted but it is what it is. Thanks!

**Comment by Tavisco** [comment]
Bump on this, I'm in the exact same position, I've bulk-imported my icloud to Immich using the CLI and the phone always try to back all up as well, 8k+ remainder assets, and yeah, I do have the "ignore icloud" setting enable, but it seems to make no difference in this case.

As a suggestion, would not be possible to checksum the photo and ask the server if it already exists BEFORE uploading it?

**Comment by shenlong-tanwen** [comment]
> As a suggestion, would not be possible to checksum the photo and ask the server if it already exists BEFORE uploading it?

That is exactly what we've planned and we have slowly started discussing on bringing this up. But since this is the core feature of the mobile app, there are a lot to be discussed.

**Reply by Tavisco** [reply]
That's nice to hear! Glad it's already being discussed :)

**Reply by DatNoHand** [reply]
Is there any update to this discussion? Having the same issue right now.

I‘ve disabled iCloud Photos on my device which deletes all local photos. Ran Immich and toggled the ignore Button. Enabled iCloud Photos again and Immich will now upload the whole iCloud Library, actively downloading photos not on device.

**Comment by pelaxa** [comment]
It would be great to be able to provide a timestamp optionally to say only import photos after this timestamp.  I just do not want everything on my phone to be backed up.

**Reply by Tavisco** [reply]
That would work as well IMHO! And should be easier to implement! Bump!

**Reply by fvillart** [reply]
Totally agree!!

**Comment by LoadingZhang** [comment]
Having the same issue, need reupload all photos even have uploaded by CLI

**Comment by chemicalsam** [comment]
Also jumping on this issue, I uploaded everything to Immich from exporting them from my Mac, but when turning it on on iOS it wants to do it again. but it seems to not respect ignore iCloud Photos.

**Comment by deluxestyle** [comment]
is there still no way to start uploading images only from "today" onwards?
